### DISCUSSION TITLE
Skip VLAN update/creation

### DISCUSSION THREAD

**Discussion by nes-achimg** [bb-Ricardo/netbox-sync#85] [Q&A]
At the moment I am testing your script against our Netbox installation. I have already done some customization regarding some custom_field configuration. But I am stuck now.
We define our VLANs outside of the Vcenter and, at the moment, those are added and updated manually. We don't want the sync script to tag, update or create any VLAN in Netbox. However, I cannot find the right place to stop the script from doing just that. I have already commented out the NBVLAN object in the Vcenter connection.py, but that doesn't have any effect on the update/creation process. I think, it only stops the initial population of the inventory.
Could you point me in the right direction?
I am looking at something like having an option in the settings.ini to switch that on or off.

**Comment by bb-Ricardo** [comment]
Hi,
You basically want to disable the option to add/sync/delete any VLAN which is reported by the vCenter? So none if this information is visible in NetBox?

I assume that is possible but would need to validate how difficult it is to implement.

**Reply by nes-achimg** [reply]
Hi Ricardo,
yes, that's right. I would be happy to assist and try to implement a solution myself. However, I find it difficult to pinpoint the part of the code where the VLAN magic happens. I thought I might be able to find something like new_vlan = self.inventory.add_update_object(NBVLAN, data...). But I haven't.

I would assume it is either in inventory.py or vmware/connection.py?

There seems to be an implicit creation of objects when those are not present in the given inventory. I had the same problem with site, though somehow solved it - we have several clusters where we don't want to have a site defined.

Going through the code again, I got an idea. Would you think, it is possible to have something like that in inventory.py?
```
    def add_object(self, object_type, data=None, read_from_netbox=False, source=None):

        # Skip object creation when it is a VLAN
        if object_type.name == "VLAN":
            return null

        # create new object
        new_object = object_type(data, read_from_netbox=read_from_netbox, inventory=self, source=source)

        # add to inventory
        self.base_structure[object_type.name].append(new_object)

        if read_from_netbox is False:
            log.info(f"Created new {new_object.name} object: {new_object.get_display_name()}")

        return new_object
```

**Reply by nes-achimg** [reply]
Hi,
it wasn't as easy as described above.
But I have eventually identified the places in the source code that needs to be modified to skip VLANs altogether.

Would it be of interest to create an issue/branch for this?

**Reply by bb-Ricardo** [reply]
Juat create a pull request and I will have a look at it if you like.

**Reply by bb-Ricardo** [reply]
Is this still required?

**Reply by nes-achimg** [reply]
Hi Ricardo. Yes, we are still very much interested in it. However, I haven't been able to pursue it lately as different tasks have been coming up.

**Reply by bb-Ricardo** [reply]
great. I also changed the VLAN "detection" code a bit around in one of the latest commits. https://github.com/bb-Ricardo/netbox-sync/blob/c26b726ba6659917fd73863f110add44141cd881/module/sources/common/source_base.py#L215

**Reply by nes-achimg** [reply]
Thanks. I will have a closer look at the code and test it in our environment. But that won't be in the next two weeks, I think.

**Reply by bb-Ricardo** [reply]
no problem

**Comment by christianahug** [comment]
Hi, 

has this been developed further? We're in the same situation where we want to manually manage our VLANs and not import them from vCenter, but I'm also having trouble identifying the relevant places in the source code.

**Reply by bb-Ricardo** [reply]
Hi, not at the moment, can you open an issue, so I can track this better?

Thank you very much.

**Comment by bb-Ricardo** [comment]
Hi @nes-achimg,

I just pushed commit "https://github.com/bb-Ricardo/netbox-sync/commit/9cfa9b192e20f61cbaeec1907ceb9191a3308ba7" to development branch. Can you check it out and test if it works for you?

Thank you

**Reply by achso18** [reply]
Hi Ricardo,
sorry for the late reply. It is holiday season...
Will do this week and give some feedback.
Thank you!

**Reply by nes-achimg** [reply]
Hi.
Eventually came around to test it in our environment. Looks really good and works flawlessly!
Thanks very much!

**Reply by bb-Ricardo** [reply]
Thank you for the heads up. Will be doing a release soon.

**Comment by nes-achimg** [comment]
Perfect. Thanks!
